xtra-libp2p-offer/src/lib.rs

```diff
@@ -1,9 +1,158 @@
+use model::olivia::BitMexPriceEventId;
+use model::ContractSymbol;
+use model::FundingRate;
+use model::Leverage;
+use model::OfferId;
+use model::OpeningFee;
+use model::Origin;
+use model::Position;
+use model::Price;
+use model::Timestamp;
+use model::TxFeeRate;
+use model::Usd;
+use serde::Deserialize;
+use serde::Serialize;
+use std::fmt;
+use time::Duration;
+
 pub mod maker;
 mod protocol;
 pub mod taker;
 
 pub const PROTOCOL: &str = "/itchysats/offer/1.0.0";
 
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
+pub struct MakerOffers {
+    pub long: Option<Offer>,
+    pub short: Option<Offer>,
+    pub tx_fee_rate: TxFeeRate,
+    pub funding_rate_long: FundingRate,
+    pub funding_rate_short: FundingRate,
+}
+
+impl fmt::Debug for MakerOffers {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("MakerOffers")
+            .field("long_order_id", &self.long.as_ref().map(|o| o.id))
+            .field("short_order_id", &self.short.as_ref().map(|o| o.id))
+            .field("tx_fee_rate", &self.tx_fee_rate)
+            .field("funding_rate_long", &self.funding_rate_long)
+            .field("funding_rate_short", &self.funding_rate_short)
+            .finish()
+    }
+}
+
+/// A concrete order created by a maker for a taker
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Offer {
```

Comment:
🔧 I would opt to introduce a trimmed down version of `Offer`, like the `OfferParams` I described elsewhere.

Reply:
For that we would need to bump the network protocol version right?

Reply:
Yep. This is part of the breaking change to the offer protocol I mentioned before.